web-ui/src/main/resources/catalog/js/edit/EditorController.js

```diff
@@ -448,14 +449,16 @@
         window.onbeforeunload = null;
 
         // if there is no history, attempt to close tab
-        if (window.history.length == 1) {
+        if ($scope.redirectUrl != null) {
```

Comment:
I guess you could check here for `$location.search()['redirectUrl']` instead of using a intermediary var on the scope

Reply:
Some custom editor may require it for some extra buttons not going through the closeEditor (at least I know one)